helper/cert/source_gen.go

```diff
@@ -182,6 +182,32 @@ func (s *GenSource) checkLeader(ctx context.Context, changed chan<- bool) {
 	}
 }
 
+func (s *GenSource) retryUpdateSecret(ctx context.Context, leaderCh chan bool, bundle Bundle) {
```

Comment:
I know we log errors as we retry, but it seems a bit odd to me to have finite retries and no error return value. From a user perspective, it would be nice to know that it gave up and restarted the whole process in between all the retry errors, so perhaps this function should return a terminal error on falling out of the retry loop, and then the `Notify.Run()` function will log it?

Although I would also be fine with infinite retries given we are cancellable through the context and leader elections. WDYT?

Reply:
Yep, I agree that does look odd without a return value, added in https://github.com/hashicorp/vault-k8s/pull/280/commits/c37fea7f6d9c50ef4fb3643125d203b3b086d9f9. I ended up changing the max time for the retries to be 90% of the certificate's life, since that's the point when the cert will be regenerated in `Certificate()`.